mir_eval/multipitch.py

```diff
@@ -0,0 +1,509 @@
+
+
+MAX_TIME = 30000.  # The maximum allowable time stamp (seconds)
+MAX_FREQ = 5000.  # The maximum allowable frequency (Hz)
+MIN_FREQ = 20.  # The minimum allowable frequency (Hz)
+
+
+def validate(ref_time, ref_freqs, est_time, est_freqs):
+    """Checks that the time and frequency inputs are well-formed.
+
+    Parameters
+    ----------
+    ref_time : np.ndarray
+        reference time stamps in seconds
+    ref_freqs : list of np.ndarray
+        reference frequencies in Hz
+    est_time : np.ndarray
+        estimate time stamps in seconds
+    est_freqs : list of np.ndarray
+        estimated frequencies in Hz
+
+    """
+
+    util.validate_events(ref_time, max_time=MAX_TIME)
+    util.validate_events(est_time, max_time=MAX_TIME)
+
+    if ref_time.size == 0:
+        warnings.warn("Reference times are empty.")
+    if ref_time.ndim != 1:
+        raise ValueError("Reference times have invalid dimension")
+    if len(ref_freqs) == 0:
+        warnings.warn("Reference frequencies are empty.")
+    if est_time.size == 0:
+        warnings.warn("Estimated times are empty.")
+    if est_time.ndim != 1:
+        raise ValueError("Estimated times have invalid dimension")
+    if len(est_freqs) == 0:
+        warnings.warn("Estimated frequencies are empty.")
+    if ref_time.size != len(ref_freqs):
+        raise ValueError('Reference times and frequencies have unequal '
+                         'lengths.')
+    if est_time.size != len(est_freqs):
+        raise ValueError('Estimate times and frequencies have unequal '
+                         'lengths.')
+
+    for freq in ref_freqs:
+        util.validate_frequencies(freq, max_freq=MAX_FREQ, min_freq=MIN_FREQ,
+                                  allow_negatives=False)
+
+    for freq in est_freqs:
+        util.validate_frequencies(freq, max_freq=MAX_FREQ, min_freq=MIN_FREQ,
+                                  allow_negatives=False)
+
+
+def resample_multipitch(times, frequencies, target_times):
+    """Resamples multipitch time series to a new timescale. Values in
+    `target_times` outside the range of `times` return no pitch estimate.
+
+    Parameters
+    ----------
+    times : np.ndarray
+        Array of time stamps
+    frequencies : list of np.ndarray
+        List of np.ndarrays of frequency values
+    target_times : np.ndarray
+        Array of target time stamps
+
+    Returns
+    -------
+    frequencies_resampled : list of lists
+        Frequency list of lists resampled to new timebase
+    """
+    if target_times.size == 0:
+        return []
+
+    if times.size == 0:
+        return [np.array([])]*len(target_times)
+
+    n_times = len(frequencies)
+
+    # scipy's interpolate doesn't handle ragged arrays. Instead, we interpolate
+    # the frequency index and then map back to the frequency values.
+    # This only works because we're using a nearest neighbor interpolator!
+    frequency_index = np.arange(0, n_times)
+
+    # assume_sorted=True for efficiency
```

Comment:
This comment is unclear to me.  Are you allowed to assume_sorted=True because the array you're interpolating is constructed above (frequency_index) in sorted order already?